docs/flashing_rust_sample.md

```diff
@@ -0,0 +1,82 @@
+# Flashing rust hello-world app
+
+## Environment prepare
+
+* Get latest stable Rust version from https://rustup.rs.
+
+* Install required target and tools
+
+```shell
+rustup target add thumbv7em-none-eabihf
+rustup component add llvm-tools-preview
+cargo install cargo-binutils
+```
+
+## Building app
+
+This is a custom app based on [blinky-button-demo](https://github.com/nrf-rs/nrf-hal/tree/master/examples/blinky-button-demo)
+sample. Code has been adapted run on nRF52840 dongle.
+
+* Download app source code
+
+TBD
+
+* Build app
+
+```shell
+cargo build --target thumbv7em-none-eabihf
+```
+
+* Convert it into binary format
+
+```shell
+cargo objdump --target thumbv7em-none-eabihf -- -O binary app.bin
+```
+
+## Flashing app
+
+Flashing is done using Olimex ARM-USB-OCD-H and OpenOCD. Note that there is an
+alternative method, using `cargo-embed` (internally uses `probe-rs`) instead but
+it does not support Olimex:
+
+```shell
+$ env RUST_LOG=trace probe-rs-cli list
+DEBUG probe_rs::probe::cmsisdap::tools > Searching for CMSIS-DAP probes using libusb
+DEBUG probe_rs::probe::cmsisdap::tools > Found 0 CMSIS-DAP probes using libusb, searching HID
+DEBUG probe_rs::probe::cmsisdap::tools > Found 0 CMSIS-DAP probes total
+DEBUG jaylink                          > libusb 1.0.23.11397
+DEBUG jaylink                          > libusb has capability API: true
+DEBUG jaylink                          > libusb has HID access: true
+DEBUG jaylink                          > libusb has hotplug support: true
+DEBUG jaylink                          > libusb can detach kernel driver: true
+No devices were found.
+```
+
+You need OpenOCD version 0.11.0 or later for flashing to work, if your distro
+has older version you need to build it from source.
+
+```shell
+sudo apt install libusb-dev libusb-1.0-0-dev libusb-1.0-0
+sudo apt install libhidapi-dev libftdi-dev libftdi1-dev
+sudo apt install libtool # for openocd
+
+git clone https://git.code.sf.net/p/openocd/code openocd
+cd openocd
+./bootstrap
+mkdir build; cd build
+../configure --enable-cmsis-dap --enable-openjtag --prefix=/opt/openocd
+make
+sudo make install
+```
+
+* To flash app, run
+
+```shell
+/opt/openocd/bin/openocd \
```

Comment:
Is it possible to have rust and desired openocd version in docker? This should go to this repo: https://github.com/fobnail/fobnail-sdk

And some very basic rust dockerfile is here: https://github.com/fobnail/fobnail-sdk/pull/2/files (it could be extended)

Reply:
@macpijan I have updated instructions to use Fobnail SDK.
SDK currently does not provide OpenOCD, I added a note that using OpenOCD requires manual installation on host.